Matveibos/hometask-3/rainfall_stat.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# this class is used for calculate mean and variance values for town rainfall
+class RainfallStat
+  SEPARATOR = ':'
+  REGULAR = /\d*\.\d/.freeze
+
+  def mean
+    create_hash
+
+    calc_mean
+  end
+
+  def variance
+    create_hash
+
+    calc_variance
+  end
+
+  private
+
+  def initialize(town, data)
```

Comment:
You don't need to move initializer into the private namespace. I left you a comment about reader methods, which you should move to the private namespace if they are not used outside of the class. So, let's move initializer back to the public namespace and create reader methods for values you need to access inside of this class